src/mesh/meshblock_pack.hpp

```diff
@@ -53,6 +55,12 @@ class MeshBlockPack {
                    const int i) const {
     return v_(block)(n)(k, j, i);
   }
+
+  KOKKOS_FORCEINLINE_FUNCTION bool IsSparseIDAllocated(const int block,
+                                                       const int var) const {
+    return v_(block).GetDim(1) > var && v_(block)(var).is_allocated();
```

Comment:
Should this be GetDim(4)? `GetDim(1)` is the `i` direction not the variable index, isn't it?
```suggestion
    return v_(block).GetDim(4) > var && v_(block)(var).is_allocated();
```

Reply:
You're right! Great catch!